internal/v1/handler_compose_image.go

```diff
@@ -180,6 +190,90 @@ func buildRepositories(arch *distribution.Architecture, imageType ImageTypes) []
 	return repositories
 }
 
+func (h *Handlers) buildRepositorySnapshots(ctx echo.Context, arch *distribution.Architecture, imageType ImageTypes, snapshotDate *string) ([]composer.Repository, error) {
+	date, err := time.Parse(time.RFC3339, *snapshotDate)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Snapshot date %s is not in RFC3339 format", *snapshotDate))
+	}
+
+	repoURLs := []string{}
+	for _, r := range arch.Repositories {
+		contains := len(r.ImageTypeTags) == 0
+		for _, it := range r.ImageTypeTags {
+			if it == string(imageType) {
+				contains = true
+				break
+			}
+		}
+		if contains {
+			repoURLs = append(repoURLs, *r.Baseurl)
+		}
+	}
+
+	repoUUIDs := []string{}
+	repoMap := map[string]content_sources.ApiRepositoryResponse{}
+	resp, err := h.server.csClient.GetRepositories(ctx.Request().Context(), repoURLs, false)
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(ctx, resp.Body)
+
+	var csRepos content_sources.ApiRepositoryCollectionResponse
+	err = json.NewDecoder(resp.Body).Decode(&csRepos)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, repo := range *csRepos.Data {
+		repoUUIDs = append(repoUUIDs, *repo.Uuid)
+		repoMap[*repo.Uuid] = repo
+	}
```

Comment:
Nil pointers, just saying. I wonder why OpenAPI fields are optional by default, that is tragic.

Reply:
So I think in the case of content sources it's fine to let it panic, as I don't expect these ever to not be set, correct @jlsherrill? We can add checks but we'd just 500 anyway I think.

Reply:
correct, they should always be set, if return a repo without a uuid something has gone tragically wrong.